packages/react-strict-dom/src/native/modules/ContextInheritedStyles.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow strict-local
+ */
+
+import type { Style } from '../../types/styles';
+
+import * as React from 'react';
+import { CSSLengthUnitValue } from '../stylex/CSSLengthUnitValue';
+import { PixelRatio, useWindowDimensions } from 'react-native';
+import { flattenStyle } from './flattenStyle';
+
+type Value = Style;
+
+type ProviderProps = $ReadOnly<{
+  children: React$MixedElement,
+  value: Value
+}>;
+
+const defaultContext = {};
+const ContextInheritedStyles: React$Context<Value> =
+  React.createContext(defaultContext);
+
+if (__DEV__) {
+  ContextInheritedStyles.displayName = 'ContextInheritedStyles';
+}
+
+// We do some special-casing to compute inherited fontSize.
+// The 'em' unit polyfill in stylex assumes the inherited fontSize is always a computed number.
+export function InheritedStylesProvider(
+  props: ProviderProps
+): React$MixedElement {
+  const { children, value } = props;
+  const inheritedStyles = useInheritedStyles();
+  const inheritedFontSize = inheritedStyles?.fontSize;
+  const valueFontSize = value?.fontSize;
+
+  const { height: viewportHeight, width: viewportWidth } =
+    useWindowDimensions();
+
+  let computedFontSize = null;
+  if (valueFontSize instanceof CSSLengthUnitValue) {
+    computedFontSize = valueFontSize.resolvePixelValue({
+      viewportHeight,
+      viewportWidth,
+      fontScale: PixelRatio.getFontScale(),
+      inheritedFontSize: inheritedFontSize
+    });
+  } else {
+    computedFontSize = valueFontSize || inheritedFontSize;
+  }
+
+  const flatStyle = flattenStyle([
+    inheritedStyles as ?Style,
+    value as ?Style,
+    computedFontSize != null && { fontSize: computedFontSize }
+  ]);
```

Comment:
This code now does 2 things:
1. consolidates inheritance of `fontSize` with the inheritable of other styles.
2. fixes a bug in the previous `fontSize` inheritance implementation that didn't polyfill nested `em` lengths correctly